### DISCUSSION TITLE
Can't run script with my .env variables

### DISCUSSION THREAD

**Discussion by Edacheclarus2** [Cyfrin/foundry-full-course-cu#377] [General]
I've been trying to run the .env variables from the terminal but its not working,

This is the error message i get 
'''
error: a value is required for '--fork-url <URL>' but none was supplied

For more information, try '--help'.
''' 
After running 
'''
forge script  script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY
'''.
It only runs if i have my private and rpc hard coded/plain text on the terminal, Someone help me out

**Comment by ChefAharoni** [comment]
Have you entered `source .env` in the terminal after adding your variables to your `.env` file?
If so, try this command:
```bash
echo $RPC_URL
```

Does this prints the $RPC_URL into the terminal? If so, then the `.env` file is working, there might be another problem.

**Reply by Edacheclarus2** [reply]
yes, i've entered source .env

echo display them blank, i don't know if its a feature of foundry to protect sensitive data or something else.
Writing in plain texts works but accessing through the variables dont

**Reply by ChefAharoni** [reply]
No, there's no such feature in Foundry. Perhaps you're calling a different name from what you've defined in the `.env` file.

This is an example of how a line should be defined in `.env`:
```env
ANVIL_RPC_URL=127.0.0.1:8545
```

These are the commands you should follow in the terminal:
```bash
source .env
echo $ANVIL_RPC_URL 
```

Then, the output will be shown like this:
```
127.0.0.1:8545
```

**Reply by Edacheclarus2** [reply]
This is exactly what I did.
I don't know why it's not working.
I can't access variable that's declared in a file.

**Reply by ChefAharoni** [reply]
Where is the .env file located? Is it within the forge folder with all other files?

In the terminal, have you CD'd on the correct folder?
One way to check it write `ls` in the terminal and see if the file is listed within other files.

**Comment by db371** [comment]
Hi, I was facing a similar issue. Here is how I mitigated it:

1. Ensure the .env file is in your root directory and not under the test. If it's in the test, you might have to cd test or just relocate the .env file to the directory by deleting and creating a new one.

2. Ensure no blank space while adding the private key and rpc url. 
For eg. PRIVATE_KEY=0xacs...
SAVE THIS FILE.

3. The command should be source .env and not source.env and you would need to re-run this prompt once again after you have made changes to the .env file. To make sure everything is smooth, use the prompt echo $PRIVATE_KEY once.

I was hardcoding the private key and URL in the terminal as well, but with these steps, I was able to finally make the environment variables work.

Hope this helps.

**Reply by Bhavik-punmiya** [reply]
Well I had the same issue whenever I ran ``` source .env ```   it said   ``` PRIVATE_KEY : command not found ```
by removing the blankspaces from the env file , it was a fix for me !!
And running ``` source .env ``` and ``` forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY ``` deployed my contract

**Reply by theirrationalone** [reply]
Are you adding data into `.env` file like this.... 

PRIVATE_KEY = 0x13j35J847
RPC_URL = https://........

If yes then you're doing it wrong ❌

Right one is... (Without white spaces before after `=`)✅

PRIVATE_KEY=0x13j35J847
RPC_URL=https://........
